src/superqt/utils/_util.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import annotations
+
+from inspect import signature
+from typing import Callable
+
+
+def get_max_args(func: Callable) -> int | None:
+    """Return the maximum number of positional arguments that func can accept."""
+    sig = signature(func)
```

Comment:
```suggestion
    try: 
	    sig = signature(func)
	except TypeError:
	    raise
	except:
	    return None
```

From documentation:

> Raises [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) if no signature can be provided, and [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) if that type of object is not supported. Also, if the annotations are stringized, and eval_str is not false, the eval() call(s) to un-stringize the annotations could potentially raise any kind of exception.

Reply:
yeah, but i really don't think _this_ is the place where that error should be raised.  I think if we can't figure out how to limit the number of arguments, we just simply shouldn't try.  we shouldn't explicitly raise an exception

Reply:
(i do agree it needed to be in a try block though!  I updated that )

Reply:
actually... on second thought, I guess the current behavior of our decorators like `ensure_main_thread` would actually hide the proper error until emit time:

```python
# on main
In [1]: from qtpy.QtCore import QObject, Signal

In [2]: class Emitter(QObject):
   ...:     sig = Signal()
   ...:

In [3]: e = Emitter()

In [4]: from superqt.utils import ensure_main_thread

In [5]: e.sig.connect(ensure_main_thread(1))
Out[5]: <PyQt6.QtCore.QMetaObject.Connection object at 0x10f09ac00>

In [6]: e.sig.emit()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File ~/dev/self/superqt/src/superqt/utils/_ensure_thread.py:93, in ensure_main_thread.<locals>._out_func.<locals>._func(*args, **kwargs)
     91 @wraps(func_)
     92 def _func(*args, **kwargs):
---> 93     return _run_in_thread(
     94         func_,
     95         QCoreApplication.instance().thread(),
     96         await_return,
     97         timeout,
     98         *args,
     99         **kwargs,
    100     )

File ~/dev/self/superqt/src/superqt/utils/_ensure_thread.py:174, in _run_in_thread(func, thread, await_return, timeout, *args, **kwargs)
    172 future = Future()  # type: ignore
    173 if thread is QThread.currentThread():
--> 174     result = func(*args, **kwargs)
    175     if not await_return:
    176         future.set_result(result)

TypeError: 'int' object is not callable
```

and this would catch that.  but I'm slightly wary of it because I've found cases before where `inspect.signature` returned a TypeError for actually callable objects (i.e. it could give us false positives)

Reply:
yeah... for example:
```python
>>> inspect.signature(print)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/talley/micromamba/envs/py38/lib/python3.8/inspect.py", line 3105, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
  File "/Users/talley/micromamba/envs/py38/lib/python3.8/inspect.py", line 2854, in from_callable
    return _signature_from_callable(obj, sigcls=cls,
  File "/Users/talley/micromamba/envs/py38/lib/python3.8/inspect.py", line 2308, in _signature_from_callable
    return _signature_from_builtin(sigcls, obj,
  File "/Users/talley/micromamba/envs/py38/lib/python3.8/inspect.py", line 2119, in _signature_from_builtin
    raise ValueError("no signature found for builtin {!r}".format(func))
ValueError: no signature found for builtin <built-in function print>
```